### PR TITLE
[After #248] Initial commit for the new ml-service API `ml_service_model_register`

### PR DESCRIPTION
**Pull request by anyj0527**
- Add callback for the new API in the daemon impl.
- Add initial impl for the C-API.
- Add a simple TC using API `ml_service_model_register`

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

### CODE REVIEW

File: tests/capi/unittest_capi_service_agent_client.cc [hunk 1]
```diff
@@ -689,6 +689,22 @@ TEST_F (MLServiceAgentTest, query_request_00_n)
   EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, status);
 }
 
+/**
+ * @brief Test the usecase of ml_servive for model. TBU.
+*/
```
Comment (songgot) [inline tests/capi/unittest_capi_service_agent_client.cc:694]: need to space

File: daemon/model-dbus-impl.cc [hunk 2]
```diff
@@ -40,10 +40,52 @@ gdbus_put_model_instance (MachinelearningServiceModel ** instance)
...
 
+/**
+ * @brief The callback function of Register method
+ *
+ * @param obj Proxy instance.
+ * @param invoc Method invocation handle.
+ * @param name The name of target model.
+ * @param path The file path of target.
+ * @return @c TRUE if the request is handled. FALSE if the service is not available.
+*/
```
Comment (songgot) [inline daemon/model-dbus-impl.cc:51]: Need a space?

**Review by songgot — APPROVED** [review]
LGTM

File: tests/capi/unittest_capi_service_agent_client.cc (hunk 1 above)
Comment (anyj0527) [reply tests/capi/unittest_capi_service_agent_client.cc]: I added a space, Thanks!

File: daemon/model-dbus-impl.cc (hunk 2 above)
Comment (anyj0527) [reply daemon/model-dbus-impl.cc]: I added a space, Thanks!

**Review by again4you — APPROVED** [review]
LGTM!

**Pull request merged**